src/wasm2js.js

```diff
@@ -1,6 +1,12 @@
 // wasm2js.js - enough of a polyfill for the WebAssembly object so that we can load
 // wasm2js code that way.
 
+// https://caniuse.com/#feat=promises
+#if MIN_CHROME_VERSION < 33 || MIN_EDGE_VERSION < 12 || MIN_FIREFOX_VERSION < 29 || MIN_IE_VERSION != TARGET_NOT_SUPPORTED || MIN_SAFARI_VERSION < 80000
+// Include a Promise polyfill for legacy browsers.
+#include "promise.js"
```

Comment:
Hmm, there are really few places where we use Promises in the runtime/preamble. Currently only place this is needed is the regular runtime preamble. Can you make this a `#if !MINIMAL_RUNTIME` for now? Since that loads up without the use of Promises. In the future if it looks like Promises are really needed, I'll look at adding that for MINIMAL_RUNTIME as well. (currently I don't think the size increase via the polyfill is warranted)

Reply:
Actually, nevermind, I realize I only have the Promise-enabled version of the default shell of MINIMAL_RUNTIME in upstream, so it makes sense to default-enable this for MINIMAL_RUNTIME as well, lgtm.

Reply:
Oh, perhaps put the comment `// https://caniuse.com/#feat=promises` inside the `#if` statement so that when that is optimized out, there won't be a stray comment `// https://caniuse.com/#feat=promises` still generated in the source (debug output deserves to look pretty as well :)